lib/ssh/src/ssh_connection_handler.erl

```diff
@@ -129,12 +129,13 @@ takeover(ConnPid, _, Socket, Options) ->
     {_, Callback, _} = ?GET_OPT(transport, Options),
     case Callback:controlling_process(Socket, ConnPid) of
         ok ->
+            Ref = erlang:monitor(process, ConnPid),
```

Comment:
Agree that the code seems to expect a monitor here. I however think it would be more natural to implement this with a gen_statem call and a server side timeout!

Reply:
I think I would like this small change to be merged and patched on supported versions.
your comment will remain relevant and should be included in planned ssh supervison tree rewrite.